### ISSUE TITLE
md-button with tooltip continues to display focus style after mouseleave

### ISSUE THREAD

**Issue by mhchen** [angular/material#4249]
This happens on master only. Normally mouseleave should cause the md-focused class to be removed, but this is now broken. It seems to have been introduced by https://github.com/angular/material/commit/e92dd26 (works before this commit) but the connection isn't immediately apparent to me.

**Comment by Frank3K** [comment]
+1

Also facing this issue. The `md-focused` class is not removed on the previously focused element. When hovering multiple buttons with a tooltip, all these button display the `md-focused` style.

**Comment by mindontheblink** [comment]
Facing a similar issue on my buttons.  It is especially noticeable on the buttons that bring up a context menu or bottom drawer.  As soon as the context menu or bottom drawer close, the trigger button is re-focused  -- sometimes with its tooltip (I have a delay set).

**Comment by karolkochan** [comment]
+1

**Comment by AleksueiR** [comment]
+1

**Comment by topherfangio** [comment]
Any chance someone could provide a quick demo showing the issue?

**Comment by karolkochan** [comment]
Demo:
http://codepen.io/karolkochan/pen/vNELLJ

First button without, socond with md-tooltip.

**Comment by topherfangio** [comment]
@karolkochan I couldn't get your demo to work (can't find ngMaterial error in console), however, I just realized that this may be the same thing as https://github.com/angular/material/issues/4225

Can you take a look and see if it's the same issue? That demo shows it pretty clearly.

**Comment by Frank3K** [comment]
@topherfangio: from what I see both demos showcase the same issue. The demo provided by @karolkochan does work for me on Firefox 40.0.0.3 and Chrome 44.0.2403.157 (both Windows 7).

**Comment by karolkochan** [comment]
@topherfangio Yes, for me it looks like duplicate (if I correctly interpreted @mhchen issue)

**Comment by TalPasi** [comment]
+1

**Comment by victor-perez** [comment]
+1

**Comment by ricardobfa** [comment]
+1

**Comment by langdonx** [comment]
This is still a bug.  Tested in Chrome 46 and Firefox 41.  Happens on master and RC1 (doesn't happen on 0.10.0)

1. https://material.angularjs.org/HEAD/demo/tooltip
2. Click refresh
3. Move mouse away from the refresh button
4. Mouse over the refresh button
5. Mouse out again... md-focused remains stuck

It seems to have something to do with focus.  If you click anywhere else after step 3, md-focused will be removed.

**Comment by erinjanae** [comment]
I'm seeing this still as well. If you click a button causing the button to gain focus then you mouse over it, the tooltip will continue to display even after you move the mouse away and are no longer hovering over it. This is happening in the https://material.angularjs.org/HEAD/demo/tooltip.

**Comment by nascosto** [comment]
Definitely still a bug

**Comment by kslo88** [comment]
Any work around for this?

**Comment by nil4you** [comment]
when this bug is going to resolved ? Even on clicking button I open a modal popup but this thing will get stuck after popup is closed.

**Comment by vdovinanton** [comment]
So, this was deprecated. Is this going to be fixed? Is there a workaround?

**Comment by cp-klewis** [comment]
I just ran into this as well. Showing a sidenav when clicking a menu button. Once the sidenav is closed the focus returns to the button clicked and the tooltip along with the ripple outline is showing. I got around this by setting the focus to the window on the click event handler then showing my sidenav. 

`$scope.toggleMenu = function (e) {
            angular.element(window).focus();
            $mdSidenav('mainMenu').toggle();
};`

**Comment by ThomasBurleson** [comment]
This issue is closed as it falls in the 'deprecated' category. We deprecate issues for one or more of the following reasons:

*  Issue has a reasonable workaround
*  Issue doesn't have a demo or reproducible case available
*  Issue is real, but fixing it is risky for users who depend on existing behavior
*  Issue is real, but only affects a very small use case
*  **Issue is real, but would require significant work to fix, and is less urgent than other needs.**

> Pull Requests are welcomed to resolve this issue. But note that **not** all community PRs will be merged into master. Pull Requests with large changes or large scope impacts will probably be rejected [due to stability risks to existing users].
